### PR TITLE
Revamp of NAFFlib to nafflib: full pythonic approach

### PR DESCRIPTION
**Pull request by pbelange**
## Description

Closes # .

## Checklist

Mandatory: 

-  I have added tests to cover my changes
-  All the tests are passing, including my new ones
- [ ] I described my changes in this PR description

Optional:

- [ ] The code I wrote follows good style practices (see [PEP 8](https://peps.python.org/pep-0008/) and [PEP 20](https://peps.python.org/pep-0020/)).
- [ ] I have updated the docs in relation to my changes, if applicable
- [ ] I have tested also GPU contexts

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,38 +1,51 @@
-from setuptools import setup, Extension, find_packages
-import sys
-import numpy as np
+# copyright ############################### #
+# This file is part of the Xpart Package.   #
```
Comment (kparasch) [inline setup.py:2]: :(

File: setup.py [hunk 2]
```diff
@@ -1,38 +1,51 @@
...
-    version="1.0.2",
-    author="Konstantinos Paraschou",
-    author_email="konstantinos.paraschou@cern.ch",
-    description="A Python-wrapped C library which implements the NAFF algorithm",
-    long_description=long_description,
-    long_description_content_type='text/markdown',
-    url="https://github.com/PyCOMPLETE/NAFFlib",
+    name='nafflib',
+    version=__version__,
+    description='nafflib algorith for frequency analysis',
```
Comment (kparasch) [inline setup.py:29]: "algorith" typo

File: nafflib/optimize.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import numba
+
+
+
+@numba.jit(nopython=True)
```
Comment (kparasch) [inline nafflib/optimize.py:6]: I am not against it, but is numba giving a significant advantage?

**Review by kparasch — APPROVED** [review]

**Review by kparasch — APPROVED** [review]

File: nafflib/optimize.py (hunk 3 above)
Comment (pbelange) [reply nafflib/optimize.py:5]: Yes, factor 2 in speed, suggested by @giadarol

**Pull request merged**